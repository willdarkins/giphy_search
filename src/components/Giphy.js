import React, { useState, useEffect } from 'react';
import axios from 'axios'

const Giphy = () => {
    //useState to store data from API
    const [content, setContent] = useState([])
    // calling giphy api
    const giphyResponse = async () => {
        const { data } = await axios.get(`https://api.giphy.com/v1/gifs/trending?api_key=LWWhHBcSHdmy5Nshmkte5jUOYsozacsy`)
        // console.log(data)
        setContent(data.data)
    }
    //passing API response through useEffect
    useEffect(() => {
        giphyResponse()
    }, [])

    return (
        // iterating over captured data from API by accessing the state variable
        <div className='container-gifs'>
            {content.map((i) => 
                <div key={i.id} className='gif'>
                    <img src={i.images.fixed_height.url} alt='gif' />
                </div>
            )}
        </div>
    )
}

export default Giphy