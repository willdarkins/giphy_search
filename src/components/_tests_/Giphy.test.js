import { render, cleanup } from '@testing-library/react';
import Giphy from '../Giphy'

describe('Giphy component', () => {
    it('renders', () => {
      render(<Giphy />);
    });
    it('matches snapshot DOM node structure', () => {
        const { asFragment } = render(<Giphy />);
        expect(asFragment()).toMatchSnapshot();
    })
  })