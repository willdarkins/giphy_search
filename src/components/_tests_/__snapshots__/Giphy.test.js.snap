// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Giphy component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <div
    class="container"
    data-testid="giphy-1"
  >
    <form
      class="form"
    >
      <input
        class="form-control"
        placeholder="Search for GIFS!!!"
        type="text"
        value=""
      />
      <button
        class="btn btn-primary mx-2"
        type="submit"
      >
        Go!
      </button>
    </form>
    <nav>
      <ul
        class="pagination pagination-sm justify-content-end border-0"
      />
    </nav>
    <div
      class="container gifs"
    >
      <div
        class="loader"
      >
        <i
          class="fas fa-spinner fa-4x fa-spin"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
